consumer.go

```diff
@@ -552,12 +562,6 @@ func (child *partitionConsumer) parseRecords(batch *RecordBatch) ([]*ConsumerMes
 	if incomplete {
 		return nil, ErrIncompleteResponse
 	}
-
-	child.offset = batch.FirstOffset + int64(batch.LastOffsetDelta) + 1
-	if child.offset <= originalOffset {
```

Comment:
Perhaps the condition needs to change, but don't we still need to account for some case where the offset doesn't advance? Or is that handled by the `incomplete` logic now?

edit: the `incomplete` logic looks like it might be wrong now anyway, I'm not sure